pkg/interface/src/views/apps/dojo/subscription.js

```diff
@@ -14,6 +14,36 @@ export default class Subscription {
     } else {
       console.error('~~~ ERROR: Must set api.ship before operation ~~~');
     }
+    this.setupSlog();
+  }
+
+  setupSlog() {
+    const slog = new EventSource('/~/slog', { withCredentials: true });
+    let available = false;
+
+    slog.onopen = e => {
+      console.log('slog: opened stream');
+      available = true;
+    }
+
+    slog.onmessage = e => {
+      this.handleEvent({ txt: e.data });
+    }
+
+    slog.onerror = e => {
+      console.error('slog: eventsource error:', e);
+      if (!available) {
+        this.handleEvent({ txt:
+          'landscape: no printf stream. bad connection or old binary.'
```

Comment:
Makes webdojo itself messier than necessary honestly. This should be a `console.log`.